src/commands/profile.ts

```diff
@@ -1,71 +1,76 @@
-import { Firestore } from '@google-cloud/firestore';
-import { Client, Message } from 'discord.js';
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { CommandInteraction, Guild, GuildMember } from 'discord.js';
+import { Command } from '../types/command';
+import { CommandBody } from '../types/command-body';
+import { CommandOptions } from '../types/command-options';
 
-exports.run = async (client: Client, msg: Message, args: string[], firestore: Firestore) => {
-  if (args === undefined || args.length === 0) {
-    const desc = module.exports.help.description;
-    const name = module.exports.help.name;
-    const usage = module.exports.help.usage;
-    msg.channel.send(`Name: ${name}\nDescription: ${desc}\nUsage: ${usage}`);
-    return;
-  }
+const data = new SlashCommandBuilder()
+	.setName('profile')
+	.setDescription('Customize your user card!')
+  .addStringOption(option =>
+    option
+      .setName('color')
+      .setDescription('Hex color code on your user profile')
+  )
+  .addStringOption(option =>
+    option
+      .setName('about')
+      .setDescription('Text on your user profile')
+  );
 
-  const userRef = firestore.collection('guilds').doc(msg.guild!.id).collection('members').doc(msg.member!.id);
+const execute: CommandBody = async (interaction: CommandInteraction, { firestore }: CommandOptions) => {
+  const guild = interaction.guild as Guild;
+  const member = interaction.member as GuildMember;
 
-  switch (args[0].toLowerCase()) {
-    case 'color':
-      if (!args[1]) {
-        msg.channel.send('Please input a valid hex color code!');
-        return;
-      }
+  const color = interaction.options.getString('color');
+  const about = interaction.options.getString('about');
 
-      let chosenColor = args[1].toUpperCase();
+  const userFirestoreRef = firestore.collection('guilds').doc(guild.id).collection('members').doc(member.id);
 
-      const isValidHex = /(^#?[0-9A-F]{6}$)|(^#?[0-9A-F]{3}$)/i.test(chosenColor);
-      if (!isValidHex) {
-        msg.channel.send('Please input a valid hex color code!');
-        return;
-      }
+  if (color) {
+    const isValidHex = /(^#?[0-9A-F]{6}$)|(^#?[0-9A-F]{3}$)/i.test(color);
+    if (!isValidHex) {
+      interaction.reply('Please input a valid hex color code!');
+      return;
+    }
 
-      const isShortenedHex = /(^#?[0-9A-F]{3}$)/i.test(chosenColor);
-      if (isShortenedHex) {
-        chosenColor = chosenColor.replace(/#?([0-9A-F])([0-9A-F])([0-9A-F])/, '$1$1$2$2$3$3');
-      }
-      userRef.set(
-        {
-          infoColor: chosenColor
-        },
-        { merge: true }
-      );
+    let newColor = color;
+    const isShortenedHex = /(^#?[0-9A-F]{3}$)/i.test(color);
```

Comment:
Looks like we have some duplicate hex code ;)

Reply:
This could probably be cleaned up and I will do that when I get back into rusty development, but as an fyi: isShortenedHex is separate from isValidHex in that isShortened hex is testing specifically for a `#XXX` pattern to know if it should be expanded to a full 6 digit code.

Reply:
will resolve in https://github.com/Mr1upMachine/rusty-bot/issues/71